awsutil/iam.go

```diff
@@ -0,0 +1,53 @@
+package awsutil
+
+import (
+  "strings"
+
+  "github.com/aws/aws-sdk-go/aws"
+  "github.com/aws/aws-sdk-go/aws/request"
+  "github.com/aws/aws-sdk-go/service/iam"
+  "github.com/aws/aws-sdk-go/service/iam/iamiface"
+  "github.com/aws/aws-sdk-go/aws/session"
+  "github.com/golang/glog"
+  "github.com/prometheus/client_golang/prometheus"
+)
+
+// ACM is our extension to AWS's ACM.acm
+type IAM struct {
+	Svc iamiface.IAMAPI
+}
+
+// NewIAM returns an IAM based off of the provided aws.Config
+func NewIAM(awsconfig *aws.Config) *IAM {
+	awsSession, err := session.NewSession(awsconfig)
+	if err != nil {
+		AWSErrorCount.With(prometheus.Labels{"service": "IAM", "request": "NewSession"}).Add(float64(1))
+		glog.Errorf("Failed to create AWS session. Error: %s.", err.Error())
+		return nil
+	}
+
+	awsSession.Handlers.Send.PushFront(func(r *request.Request) {
+		AWSRequest.With(prometheus.Labels{"service": r.ClientInfo.ServiceName, "operation": r.Operation.Name}).Add(float64(1))
+		if AWSDebug {
+			glog.Infof("Request: %s/%s, Payload: %s", r.ClientInfo.ServiceName, r.Operation, r.Params)
+		}
+	})
+
+	iamClient := IAM{
+		iam.New(awsSession),
+	}
+	return &iamClient
+}
+
+// CertExists checks whether the provided ARN existing in AWS.
+func (i *IAM) CertExists(arn *string) bool {
+  arn_string := *arn
+  certificate_name := arn_string[strings.LastIndex(arn_string, "/") + 1:len(arn_string)]
```

Comment:
Switch vars to mixed case or shorter vars rather than using `_`: https://golang.org/doc/effective_go.html#mixed-caps

e.g. `arn_string` -> `a` ... `certificate_name` -> `c`